### PR TITLE
dislocker 0.7.1 (new formula)

### PR DESCRIPTION
**Pull request by DocLM**
-  [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by woodruffw** [comment]
@BrewTestBot test this please

**Comment by woodruffw** [comment]
```
Error: 1 problem in 1 formula
dislocker:
  * Non-libraries were installed to "/usr/local/opt/dislocker/lib"
    Installing non-libraries to "lib" is discouraged.
    The offending files are:
      /usr/local/opt/dislocker/lib/libdislocker.bundle
```

File: Formula/dislocker.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  homepage "https://github.com/Aorimn/dislocker"
+  url "https://github.com/Aorimn/dislocker/archive/v0.7.1.tar.gz"
+  sha256 "742fb5c1b3ff540368ced54c29eae8b488ae5a5fcaca092947e17c2d358a6762"
+
+  bottle do
+    sha256 "a7af4bb7e8615632eb6dfd0e74dbfbb7933f98f1723e588e24bd9e7d2621a770" => :high_sierra
+  end
+
+  depends_on "cmake" => :build
+  depends_on "ruby@1.8" => :build
```
Comment (DomT4) [inline Formula/dislocker.rb]: It seriously only builds using Ruby 1.8?

Comment (DomT4) [reply Formula/dislocker.rb]: Pretty sure this wouldn't just be a `:build` dep either?

File: Formula/dislocker.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class Dislocker < Formula
+  desc "FUSE driver to read/write Windows' BitLocker-ed volumes"
+  homepage "https://github.com/Aorimn/dislocker"
+  url "https://github.com/Aorimn/dislocker/archive/v0.7.1.tar.gz"
+  sha256 "742fb5c1b3ff540368ced54c29eae8b488ae5a5fcaca092947e17c2d358a6762"
+
+  bottle do
```
Comment (DomT4) [inline Formula/dislocker.rb]: Don't add `bottle do` blocks. CI will handle that as necessary.

Comment (DocLM) [reply Formula/dislocker.rb]: Removed

File: Formula/dislocker.rb (hunk 1 above)
Comment (DocLM) [reply Formula/dislocker.rb]: I found a way to disable ruby dependency

File: Formula/dislocker.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on :osxfuse
+
+  def install
+    system "cmake", "-DCMAKE_DISABLE_FIND_PACKAGE_Ruby=TRUE", *std_cmake_args
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/dislocker", "-h"
```
Comment (DomT4) [inline Formula/dislocker.rb]: Can _anything_ be tested here in terms of functionality? Possibly not, given it needs `fuse`, but. Ideally.

Comment (DocLM) [reply Formula/dislocker.rb]: No, I don't think because it need a bitlocker encrypted volume.

**Comment by DomT4** [comment]
The CI has wandered off here. Can a maintainer ask @BrewTestBot to do its thing again, Ta.

**Comment by ilovezfs** [comment]
Well it needs a real test so I'm not sure why we're re-running CI again but sure.

@BrewTestBot test this please

**Comment by DomT4** [comment]
I couldn't remember if the test was the only failure, and I didn't want to be like hey we should fix the test only to discover the build was erupting 🙃.

**Comment by DomT4** [comment]
I mean I could've just pulled locally & tested, but it's 06:50 & my brain is like ☠️.

**Comment by ilovezfs** [comment]
```
Error: 1 problem in 1 formula
dislocker:
  * Non-libraries were installed to "/usr/local/opt/dislocker/lib"
    Installing non-libraries to "lib" is discouraged.
    The offending files are:
      /usr/local/opt/dislocker/lib/libdislocker.bundle
```

**Comment by DomT4** [comment]
Cool. Apparently `-DCMAKE_DISABLE_FIND_PACKAGE_Ruby=TRUE` ain't doing a lot.

The installation location should be something like `lib/"ruby/#{ruby_version}/dislocker`, IIRC. You should be able to pass this to `cmake` somewhere, or will need to find a variable to disable Ruby it actually obeys.

In terms of testing, the most plausible route to go down seems to be to spoof a `.bek` file and use `dislocker-bek` to prod it. Is that doable or just a nightmare? I barely touch Windows these days so 🤷‍♂️.

**Comment by DocLM** [comment]
@DomT4 why did you say that `-DCMAKE_DISABLE_FIND_PACKAGE_Ruby=TRUE` does not disable ruby?

**Comment by DomT4** [comment]
> why did you say that -DCMAKE_DISABLE_FIND_PACKAGE_Ruby=TRUE does not disable ruby?

To be honest I didn't check, but the most frequent _valid_ use of `.bundle`s on macOS are Ruby extensions.

**Comment by DocLM** [comment]
I made two build log, one with the option and the other without, also commented the part of src/CMakeList.txt where it rename the bundle file (now it is dislocker_bundle.so).
[with_ruby.log](https://github.com/Homebrew/homebrew-core/files/1408429/with_ruby.log)
[without_ruby.log](https://github.com/Homebrew/homebrew-core/files/1408430/without_ruby.log)

> -DCMAKE_DISABLE_FIND_PACKAGE_Ruby=TRUE

you can see from the log that with this variable set it does not compile ruby related stuff

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Ignoring the audit.

**Comment by ilovezfs** [comment]
```
dislocker:
  * Non-libraries were installed to "/usr/local/opt/dislocker/lib"
    Installing non-libraries to "lib" is discouraged.
    The offending files are:
      /usr/local/opt/dislocker/lib/libdislocker.bundle
```

**Comment by ilovezfs** [comment]
Thanks for the pull request @DocLM and your first contribution to Homebrew! Shipped.

**Pull request closed without merging**